scripts-python/case.build

```diff
@@ -0,0 +1,508 @@
+#!/usr/bin/env python
+
+"""
+Script to build a case.
+"""
+
+from standard_script_setup import *
+
+from CIME.utils import expect, run_cmd
+from CIME.case import Case
+from CIME.env_module import EnvModule
+
+import argparse, doctest, shutil, glob
+
+###############################################################################
+def parse_command_line(args, description):
+###############################################################################
+    parser = argparse.ArgumentParser(
+        usage="""\n%s [<casedir>] [--verbose]
+OR
+%s --help
+OR
+%s --test
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# Build case \033[0m
+    > %s
+""" % ((os.path.basename(args[0]), ) * 4),
+
+description=description,
+
+formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    parser.add_argument("caseroot", nargs="?", default=os.getcwd(),
+                        help="Case directory to build")
+
+    parser.add_argument("-t", "--testmode", action="store_true",
+                        help="Unknown TODO")
```

Comment:
The idea is that we want to be able to run this script on machines other than the machine specified for the build and
so in testmode it should ignore certain errors such as input or output paths.   It's not been implemented consistently across scripts, but it is a feature that I want to support in some form.   An example of where this is useful:  using the --namelist-only flag to create_test I can test for namelist changes on any machine from a single location.